### ISSUE TITLE
Build broken on master after espree upgrade

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#11014]
**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```bash
npm test
```

**What did you expect to happen?**

The build should pass.

**What actually happened? Please include the actual, raw output from ESLint.**

The build did not pass. The following output was produced during the gensite tests:

```
PhantomJS 2.1.1 (Mac OS X 0.0.0) ERROR
  {
    "message": "SyntaxError: Expected an identifier but found '{' instead\nat build/eslint.js:7581:0",
    "str": "SyntaxError: Expected an identifier but found '{' instead\nat build/eslint.js:7581:0"
  }

Finished in 0.174 secs / 0 secs @ 16:47:26 GMT-0400 (Eastern Daylight Time)

SUMMARY:
✔ 0 tests completed
PhantomJS 2.1.1 (Mac OS X 0.0.0) ERROR
  {
    "message": "TypeError: undefined is not a constructor (evaluating 'new Linter()')\nat tests/lib/linter.js:2534:28",
    "str": "TypeError: undefined is not a constructor (evaluating 'new Linter()')\nat tests/lib/linter.js:2534:28"
  }

✖ Error while running the tests! Exit code: 1
```

I haven't investigated the problem yet, but I found that downgrading `espree` back to `4.0.0` made the problem go away. At first glance, it seems like the generated browser version of `Linter` has a syntax error (or ES6 syntax).

**Comment by not-an-aardvark** [comment]
The problem might be that https://github.com/eslint/espree/commit/8eadb8829e6ca55dced35243665a6e19b7ebf386 added ES6 syntax to `espree` for the first time, and we're not compiling code in dependencies down to ES5. As far as semver is concerned, this is valid since `espree` supports Node `>=6.0.0`, but if this is the cause of the issue we might need to adjust our build process so that we also compile dependencies.
